### PR TITLE
Update eveonline examples for v4

### PR DESCRIPTION
**Pull request by mckernanin**
## Changes 💡
- update eveonline examples for v4

### CODE REVIEW

**Comment by balazsorban44** [comment]
Thanks, @mckernanin! A related question: Assuming you are using EVEOnline and tried v4, were you able to use it with v4, or you needed to modify the default config (https://github.com/nextauthjs/next-auth/blob/beta/src/providers/eveonline.js)?

File: docs/providers/eveonline.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ options: {
       }
       return token;
     },
-    session: async (session, token) => {
+    session: async ({ session, token }) => {
       if (token) {
         session.user.id = token.id;
       }
```
Comment (balazsorban44) [inline docs/providers/eveonline.md:59]: You can also drop the if statement here.
```suggestion
    session: async ({ session, token }) => {
      session.user.id = token.id;
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Strictly speaking, that whole section about setting the id is irrelevant/not specific for this Provider, and we already document (https://next-auth.js.org/configuration/callbacks) how to extend the session with additional properties. I might actually just remove that whole section to be honest.

**Comment by mckernanin** [comment]
Thanks for the review @balazsorban44 I'll check it out. The EVE dev portal for getting tokens has been busted for the last week, they fixed it today so I was able to get updated creds. Working through getting it up and running, I'm getting an error about requested scopes. I'll raise an issue on the main repo.

File: docs/providers/eveonline.md (hunk 1 above)
Comment (mckernanin) [reply docs/providers/eveonline.md:59]: Noice. Didn't want to completely rewrite away from the initial style. Once I get the EVE stuff up and running I'd be happy to update this further.

**Comment by balazsorban44** [comment]
Be aware of this issue before opening a new one! https://github.com/nextauthjs/next-auth/issues/2524

**Comment by mckernanin** [comment]
@balazsorban44 I tested and reported my findings on the main repo: https://github.com/nextauthjs/next-auth/issues/2524#issuecomment-919782490

I opened a PR on the main repo as well, to get it working with v4: https://github.com/nextauthjs/next-auth/pull/2759

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thank you!

**Pull request merged**